integrations/slack/src/misc/utils.ts

```diff
@@ -131,9 +131,42 @@ export const getSlackTarget = (conversation: Conversation) => {
   return { channel, thread_ts: thread }
 }
 
-export async function sendSlackMessage(botToken: string, ack: AckFunction, payload: ChatPostMessageArguments) {
-  const client = new WebClient(botToken)
-  const response = await client.chat.postMessage(payload)
+const isValidUrl = (str: string) => {
+  try {
+    new URL(str)
+    return true
+  } catch (err) {
+    return false
+  }
+}
+
+const getOptionalProps = (ctx: IntegrationCtx, logger: IntegrationLogger) => {
+  if (ctx.configuration.botAvatarUrl && isValidUrl(ctx.configuration.botAvatarUrl)) {
+    return {
+      username: ctx.configuration.botName,
+      icon_url: ctx.configuration.botAvatarUrl,
+    }
+  }
+
+  if (ctx.configuration.botAvatarUrl && !isValidUrl(ctx.configuration.botAvatarUrl)) {
```

Comment:
Due to the `return` above we could skip repeating the URL check

```suggestion
  if (ctx.configuration.botAvatarUrl) {
```

However to keep this clear/concise we could just nest the `isValidUrl` check inside the `if (ctx.configuration.botAvatarUrl)` and wrap the `.warn()` in an `else`.

Reply:
OK no worries! this is more of a code style thing anyway

Reply:
It's just that repetition activates my OCD 😆